### PR TITLE
No more complaining about FIFO's being empty (addresses #1147)

### LINKED ISSUE #1147: kfile.check_valid_file_exists() should return silently if it's a FIFO
see logic in kfile.check_input_files() for example.

As it is, running `normalize-by-median` on a FIFO warns that the FIFO is empty.

### PR DESCRIPTION
**Pull request by bocajnotnef**
fixes #1147 

- [x] Is it mergeable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Is the Copyright year up to date?

@ctb @luizirber CR/merge please.

### CODE REVIEW

**Comment by mr-c** [comment]
Needs a test

**Comment by bocajnotnef** [comment]
@mr-c Updated

File: khmer/kfile.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def check_valid_file_exists(in_files):
     """
     for in_file in in_files:
         if os.path.exists(in_file):
-            if os.stat(in_file).st_size > 0:
+            if os.stat(in_file).st_size >= 0:
```
Comment (ctb) [inline khmer/kfile.py]: am I missing something? if the file size is 0 this will not print a warning, right? don't we want the warning in that case?

**Comment by ctb** [comment]
I don't understand this pull request at all.

- the fix doesn't address FIFOs, and seems to apply to all empty files;
- there's no test that breaks when the "fix" to kfile.py is applied;
- the new test doesn't break when I revert the change to kfile;

So,

- write a test that breaks with your current changes to kfile.py line 152.
- revert that change, which is a bad one.
- please address FIFO size specifically, as in #1147.

Or, alternatively, tell me that I'm wrong and convince me :)

**Comment by bocajnotnef** [comment]
Misunderstood how stat works in this case. I'll sort it.

**Comment by bocajnotnef** [comment]
@ctb updated. pep8 and tests passing locally, test breaks when change is reverted and kfile still complains about empty files. 


Jenkins linux build appears to be hanging/breaking for other reasons.

**Comment by ctb** [comment]
OK - could you please 

* write a test that breaks with your previous changes to kfile.py line 152

as well?

**Comment by ctb** [comment]
(the current code looks good, although if you wanted to use parantheses to put an explicit grouping around the 'and not' clause in the if statement, it would make it easier to read.)

**Comment by bocajnotnef** [comment]
As in, the bad changes I made? (Allowing all empty files through)

**Comment by ctb** [comment]
yes.

**Comment by bocajnotnef** [comment]
tests/pep8 passing locally--jenkins is borked for other reasons

@ctb Updated; Merge, please?

**Comment by mr-c** [comment]
@bocajnotnef is this urgent? I'd rather fix Jenkins right now.

**Comment by bocajnotnef** [comment]
@mr-c Nope, is not urgent.

**Pull request merged**